### ISSUE TITLE
sits_train only using 1 core

### ISSUE THREAD

**Issue by LorisDeBiasi** [e-sensing/sits#885]
**Describe the bug**
Hello. I'm facing an issue with sits_train. I don't know why but it only uses 1 core.

I didn't find a "multicore" (or something similar) parameter like some other functions and from what I've read, it should by default use all the core available (at least for the torch models https://github.com/e-sensing/sits/issues/653)

**To Reproduce**
Here is a minimal working example :

```
library(sits)
library(sitsdata)

tempcnn_model <- sits_train(
  samples_matogrosso_mod13q1,
  sits_tempcnn()
)
```

**Additional context**
The cpu usage :
![sits_train_core](https://user-images.githubusercontent.com/7312028/208661143-9ae6e64f-0c0e-4bc6-acde-97c59080144f.jpg)

The sessionInfo() :
```
R version 4.2.1 (2022-06-23)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /opt/ebsofts/FlexiBLAS/3.2.0-GCC-11.3.0/lib64/libflexiblas.so.3.2

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C
 [9] LC_ADDRESS=C               LC_TELEPHONE=C
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base

other attached packages:
[1] sitsdata_0.4.2 sits_1.2.0

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.8.3      pillar_1.8.1      compiler_4.2.1    prettyunits_1.1.1
 [5] progress_1.2.2    torchopt_0.1.2    zeallot_0.1.0     bit_4.0.4
 [9] lubridate_1.8.0   lifecycle_1.0.3   tibble_3.1.8      pkgconfig_2.0.3
[13] rlang_1.0.6       DBI_1.1.3         cli_3.4.1         yaml_2.3.5
[17] coro_1.0.3        withr_2.5.0       dplyr_1.0.9       generics_0.1.2
[21] vctrs_0.5.1       fs_1.5.2          hms_1.1.1         bit64_4.0.5
[25] tidyselect_1.1.2  glue_1.6.2        data.table_1.14.2 R6_2.5.1
[29] processx_3.6.1    fansi_1.0.3       luz_0.3.1         tidyr_1.2.0
[33] callr_3.7.0       purrr_0.3.4       magrittr_2.0.3    ps_1.7.1
[37] ellipsis_0.3.2    assertthat_0.2.1  torch_0.9.0       utf8_1.2.2
[41] crayon_1.5.1
```

**Comment by rolfsimoes** [comment]
We will implement the multicores parameter in our torch models to enable multithreading functionality. This will allow the models to utilize multiple cores of a CPU in parallel, potentially improving the performance of certain tasks.

In previous versions of our models, we had to remove this feature due to issues arising in functions that required multitasking capabilities such as cross-validation and hyperparameter tuning. However, we have addressed these issues and can now reintegrate the multicores parameter without encountering errors.
